controllers/trivy_vuln_crds_controller.go

```diff
@@ -0,0 +1,219 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strconv"
+)
+
+const LagoonProjectLabel = "lagoon.sh/project"
+const TrivyVulnReportNamespace = "trivy-operator.resource.namespace"
+
+type TrivyVulnerabilityReportReconciler struct {
+	Client         client.Client
+	Scheme         *runtime.Scheme
+	MessageQWriter func(data []byte) error
+	WriteToQueue   bool
+	PredicateData  interface{}
+}
+
+type ProblemsPayload struct {
+	Problems []interface{} `json:"problems"`
+}
+
+// predicate that filters out non-lagoon projects
+func (r *TrivyVulnerabilityReportReconciler) lagoonProjectLabelsOnlyPredicate() predicate.Predicate {
+
+	return predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			if r.isLagoonProject(event.Object) &&
+				!insightsProcessedAnnotationExists(event.Object) {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			if r.isLagoonProject(event.ObjectNew) &&
+				!insightsProcessedAnnotationExists(event.ObjectNew) {
```

Comment:
More a question than a comment - under what circumstances does a trivy report change?
Does the operator create _new_ reports for each scan rather than update?

Reply:
This is something I wanted to discuss since I _believe both options are available. I.e. The operator can be configured to create new reports for each scan rather than updating existing reports (useful for tracking changes over time but will also result in a large number of reports being generated). Or, it can be configured to update existing reports.